### PR TITLE
implement dataset metadata builder

### PR DESCRIPTION
**Pull request by atarashansky**
This code was copied from the original WMG pipeline with no changes into a separate module.

### CODE REVIEW

File: backend/wmg/pipeline/dataset_metadata.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+@log_func_runtime
+def create_dataset_metadata(*, corpus_path: str) -> None:
+    """
+    This function generates a dictionary containing metadata for each dataset.
+    The metadata includes the dataset id, label, collection id, and collection label.
+    The function fetches the datasets from the curation API and iterates over them to create the metadata dictionary.
```
Comment (atolopko-czi) [inline backend/wmg/pipeline/dataset_metadata.py:17]:
```suggestion
    The function fetches the datasets from the Discover API and iterates over them to create the metadata dictionary.
```

File: backend/wmg/pipeline/dataset_metadata.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    logger.info("Generating dataset metadata file")
+    pipeline_state = load_pipeline_state(corpus_path)
+    datasets = get_datasets_from_curation_api()
```
Comment (atolopko-czi) [inline backend/wmg/pipeline/dataset_metadata.py:21]: s/curation/discover/

File: backend/wmg/pipeline/dataset_metadata.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+@log_func_runtime
+def create_dataset_metadata(*, corpus_path: str) -> None:
+    """
+    This function generates a dictionary containing metadata for each dataset.
+    The metadata includes the dataset id, label, collection id, and collection label.
+    The function fetches the datasets from the curation API and iterates over them to create the metadata dictionary.
+    """
+    logger.info("Generating dataset metadata file")
+    pipeline_state = load_pipeline_state(corpus_path)
```
Comment (atolopko-czi) [inline backend/wmg/pipeline/dataset_metadata.py:20]: any need to verify the expected state first?

Comment (atarashansky) [reply backend/wmg/pipeline/dataset_metadata.py:20]: `dataset_metadata` is not dependent on any previous steps.

**Pull request merged**